enterprise/security/src/main/java/org/neo4j/server/security/enterprise/auth/AbstractRoleRepository.java

```diff
@@ -96,7 +96,8 @@ public void update( RoleRecord existingRole, RoleRecord updatedRole )
         // Assert input is ok
         if ( !existingRole.name().equals( updatedRole.name() ) )
         {
-            throw new IllegalArgumentException( "updated role has a different name" );
+            throw new IllegalArgumentException( "Changing a role's name is not allowed. Change attempted from name '" +
+                    existingRole.name() + "' to '" + updatedRole.name() + "'." );
```

Comment:
Much better 👍 
I would suggest only the following small change to the whole text:
"The attempt to update the role from 'ABC' to 'PQR' failed. Changes to roles are not allowed." (I think we can leave out the concept of a role _name_ here, as it is superfluous). I hesitated over whether to say something about roles not being modifiable _in this version_, but that may not be a good idea!